### DISCUSSION TITLE
Styled-components - styles disappear when rehydrating

### DISCUSSION THREAD

**Discussion by IvanDmitrachkov** [vercel/next.js#43544] [Help] [answered]
when I use styled components, install like it docs https://styled-components.com/docs/advanced#nextjs, styles disappear when rehydrating, and Server Side DOM differs from original dom

example
server side
```
<html>
<body>
<!--StartFragment-->

<style data-styled="" data-styled-version="5.3.6">
--
*{box-sizing:border-box;word-wrap:break-word;}/*!sc*/
body{font-family:Arial,Helvetica,Verdana,sans-serif;font-size:16px;font-weight:normal;-webkit-letter-spacing:.03rem;-moz-letter-spacing:.03rem;-ms-letter-spacing:.03rem;letter-spacing:.03rem;margin:0 auto;}/*!sc*/
h1{font-size:4rem;}/*!sc*/
a{color:#bf9e5f;-webkit-text-decoration:none;text-decoration:none;cursor:pointer;}/*!sc*/
a:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
img{border:0px;width:100%;}/*!sc*/
data-styled.g1[id="sc-global-hKibkQ1"]{content:"sc-global-hKibkQ1,"}/*!sc*/
.dGKnqg{max-width:1200px;margin:0 auto;overflow:hidden;text-align:center;}/*!sc*/
data-styled.g2[id="pages__Container-sc-1snu5ue-0"]{content:"dGKnqg,"}/*!sc*/
</style>

<!--EndFragment-->
</body>
</html>
```

client side 

```
<style data-styled="active" data-styled-version="5.3.6"></style>
```

example project https://github.com/koalamango/nextjs-styled-components-starter

**Comment by IvanDmitrachkov** [comment]
page view is ok, but it problem for SEO

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Since the initial HTML contains the style tag with the Styled Components, **the SEO is pretty much out of harms way**!

If I use another type of client, instead of the browser, or if I disable JS I still see the correct styles. 

But, what's going on we may ask? Why do we see:

```html
<style data-styled="active" data-styled-version="5.3.6"></style>
```

Well... there's something called the CSSOM, take for example:

![Screenshot 2022-11-29 at 21 39 40](https://user-images.githubusercontent.com/21013447/204642839-c4426757-230d-49e4-baa3-7861c4d2af66.png)

Which is an optimization made by the framework to dynamically inject styles as we do.

However, this is a subject which is best explained in this post https://www.joshwcomeau.com/react/demystifying-styled-components/, please do read through it!

**Reply by IvanDmitrachkov** [reply]
thank you very much!
